core/middleware.py

```diff
@@ -15,3 +18,26 @@ def __call__(self, request):
             request.axes_attempt_time = now()
 
         return self.get_response(request)
+
+
+class GraphQLRateLimitMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        group = settings.RATELIMIT_GROUP
+        key = settings.RATELIMIT_KEY
+        rate = settings.RATELIMIT_RATE
+        if request.path == '/api/graphql':
+            rate_limited = is_ratelimited(
+                request=request,
+                group=group,
+                key=key,
+                rate=rate,
+                method=is_ratelimited.ALL,
+                increment=True
+            )
+            if rate_limited:
+                return JsonResponse({'detail': 'Rate limit exceeded'}, status=429)
```

Comment:
I'm not sure if we should treat all requests in same manner but on the other hand we don't have any identified actors that should have higher rates. Have you tested that default limits are sufficient to use application without interruptions?

Reply:
So the case is that it is very unlikely to have the DoS attack by authenticate user so we have a lockout mechanism for login attempts and a high number for requests per minute, which is 150/m. I've tested smaller number like 25/50 etc. and it was not enough since entering insuree page may require 15-25 requests...

I've tested it by going from page to page and trying to count requests made.